tests/test_modeling_common.py

```diff
@@ -3597,238 +3622,243 @@ def test_eager_matches_sdpa_inference(self, torch_dtype: str):
             ("cuda", True, torch.float16): 5e-3,
         }
 
-        def get_mean_reldiff(failcase, x, ref, atol, rtol):
-            return f"{failcase}: mean relative difference: {((x - ref).abs() / (ref.abs() + 1e-12)).mean():.3e}, torch atol = {atol}, torch rtol = {rtol}"
-
         set_model_tester_for_less_flaky_test(self)
 
         for model_class in self.all_model_classes:
             config, inputs_dict = self.model_tester.prepare_config_and_inputs_for_common()
             set_config_for_less_flaky_test(config)
             model = model_class(config)
-            # FIXME: we deactivate boolean mask for models using "use_mask_token" in their constructors.
-            # These models support masking only in the case `use_mask_token=True`. Otherwise they cannot consume an input mask.
-            # This means that the class needs to be instantiated much later, after `use_mask` is set, which means a significant refactor of the code.
-            # However masking there is not done at any layers that matters (i.e self-attention), therefore we can safely deactivate it.
-            deactivate_mask = "use_mask_token" in inspect.signature(model_class).parameters
-            is_encoder_decoder = model.config.is_encoder_decoder
+            # TODO: standardize the interfaces for musicgen models, see other todo in this test
+            if model.__class__.__name__ == "MusicgenMelodyForConditionalGeneration":
```

Comment:
do you mean we need this because of the new generate tests have new names?

Reply:
maybe I misunderstand: the change here is to avoid this test being overwitten for musicgen models?

Reply:
No :) For this specific model, which behaves differently, we have two usual options:
- overwrite the test as usual
- add this exception

I went with the second to pressure me into having a second look at musicgen, which has many interface issues (same argument names, different meaning and expected shapes) 👀 These interface issues, in turn, are causing me issues whenever we fix `generate`, tests, ...

I can do a normal test overwrite if you think it's preferable.

Reply:
It's ok, I have no strong opinion, just wondering if : with this `if ... else ...`, `musicgen models` is still run this test successfully without the `overwrite`. If so, that is nice.

Reply:
Yes, they pass the test 🙌